src/Renderer.js

```diff
@@ -55,6 +55,73 @@ export default class Renderer {
     return nodeEl;
   }
 
+  // extract all the literals, create clones, and absolutely position 
+  // them at their original locations
+  animateTransition(ast, toBlocks) {
+    var that = this, rootNodes = ast.rootNodes;
+    // take note of the parent elt, CM offsets, and literals from the AST
+    var cm = this.cm, parent = this.cm.getScrollerElement();
+    let {left: offsetLeft, top: offsetTop} = parent.getBoundingClientRect();
+
+    // toDom : AST Node -> DOM Node
+    // given a literal AST node, make a DOM node with the same srcLoc info
+    function toDom(literal) {
+      let el = document.createElement("span");
+      el.appendChild(document.createTextNode(literal.value.toString()));
+      return el;
+    }
+
+    // given a literal, a clone, and whether we're coming from text...
+    // position the clone over the currently-rendered literal
+    function assignClonePosition(literal, clone, fromText) {
+      if(fromText){ // if we're coming from text, fake a literal to get coords
+        literal.el = toDom(literal);
+        var tm = cm.markText(literal.from, literal.to, { replacedWith: literal.el });
+      }
+      // assign the location and other style info
+      let {left, top, width, height} = literal.el.getBoundingClientRect(); 
+      clone.style.top    = (top - offsetTop) + parent.scrollTop  + "px";
+      clone.style.left   = (left- offsetLeft)+ parent.scrollLeft + "px";
+      clone.style.width      = width  + "px";
+      clone.style.height     = height + "px";
+      clone.style.display    = "inline-block";
+      clone.style.position   = "absolute";
+      clone.style.animation  = "none";
+      clone.className        = "transition";
+      if(fromText) { tm.clear(); } // clean up the faked marker
+    }
+
+    // 1) get all the literals from the AST, and make clones of them
+    let literals = rootNodes.reduce((acc, r) => {
+      return acc.concat(Array.from(r).filter((n) => n.type==="literal"));
+    }, []);
```

Comment:
since your reduce function only has one line in it, you can ditch the curly braces. i.e.

```
let literals = rootNodes.reduce((acc, r) => acc.concat(Array.from(r).filter((n) => n.type==="literal")), []);
```